src/main.js

```diff
@@ -0,0 +1,79 @@
+var saveButton = document.querySelector('#save');
+var titleInput = document.querySelector('#titleInput');
+var bodyInput = document.querySelector('#bodyInput');
+var ideaParent = document.querySelector('#ideaParent');
+
+var savedIdeas = [];
+
+
+saveButton.addEventListener('click', createIdeaCard);
+window.addEventListener('load', checkInputFields);
+titleInput.addEventListener('keyup', checkInputFields);
+bodyInput.addEventListener('keyup', checkInputFields);
+
+
+function createIdeaCard() {
+  if (titleInput.value !== "" && bodyInput.value !== "") {
+    var newIdea = new Idea();
+    newIdea = createNewIdea();
+    updateIdeaArray(newIdea);
+    renderIdeaCard();
+    clearInputFields();
+  }
+}
+
+function createNewIdea() {
+  var newIdea = new Idea(titleInput.value, bodyInput.value);
+  //newIdea.generateRdmId(savedIdeas);
+  return newIdea;
+}
+
+function updateIdeaArray(newIdea) {
+    savedIdeas.push(newIdea);
+  }
+
+function renderIdeaCard() {
+  ideaParent.innerHTML = "";
+  for (var i = 0; i < savedIdeas.length; i++) {
+    ideaParent.innerHTML +=
+    `<article class="ideas-display" id="ideasGrid">
+      <section class="idea-header">
+        <img class="idea-favorite" id="favoriteButton" src="./assets/star.svg" alt="A star for favoriting">
+        <img class="delete-idea" id="deleteButton" src="./assets/delete.svg" alt="An X for deleting ideas">
+      </section>
+      <section class="idea-card-inner">
+        <h3 class="idea-title">${savedIdeas[i].title}</h3>
+        <p class="idea-body">${savedIdeas[i].body}</p>
+      </section>
+      <section class="idea-box-footer">
+        <img class="idea-comment-img" id="commentButton" src="./assets/comment.svg" alt="A comment button">
+        <p class="idea-comment">Comment</p>
+      </section>
+    </article>`
+  }
+}
+
+function clearInputFields() {
+  titleInput.value  = "";
+  bodyInput.value = "";
+}
+
+function checkInputFields() {
+  if (titleInput.value !== "" && bodyInput.value !== "") {
+    saveButton.disabled = false;
+    removeButtonState();
+  }else{
+    saveButton.disabled = true;
+    addButtonState();
+  }
+}
+
+function addButtonState() {
+  saveButton.classList.add("disabled-button");
+  saveButton.innerHTML = "Enter valid idea";
```

Comment:
What quantifies a valid idea for your application? For consumer empathy, additional instruction on exactly what is expected goes a long way.